src/controllers/SocialsController.js

```diff
@@ -0,0 +1,174 @@
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable camelcase */
+import models from '../db/models';
+import TokenAuthenticate from '../helpers/TokenAuthenticate';
+import Response from '../helpers/response';
+
+const { User, Profile } = models;
+const tokenExpireTime = '10hr';
+let response;
+/** SocialController Class */
+class SocialController {
+  /**
+   * @static
+   *
+   * @param {object} accessToken
+   * @param {object} refreshToken
+   * @param {object} profile - information from facebook
+   * @param {*} done - returns the user object
+   * @returns {*} - passes execution to next middleware on route path
+   * @memberof SocialAuthController
+   */
+  static facebookSignin(accessToken, refreshToken, profile, done) {
+    const {
+      first_name, middle_name, email
+    } = profile._json;
+    const { bio } = profile._json || null;
+    const { imgURL } = profile._json || null;
+    User.findOrCreate({
+      where: { email },
+      defaults: {
+        username: `${first_name} ${middle_name}`,
+        email,
+        signUpType: 'social',
+        userType: null,
+        isVerified: true,
+        password: null,
+      }
+    }).spread((user, isCreated) => {
+      const userDetails = user.dataValues;
+      if (isCreated) {
+        Profile.create({
+          userId: userDetails.id,
+          firstName: first_name,
+          lastName: middle_name,
+          bio,
+          imgURL,
+        });
+      }
+      userDetails.newUser = isCreated;
+      return done(null, userDetails);
```

Comment:
Why are you using a callback? It's required by passport?

Reply:
yes it is